### PR TITLE
refactor: 💡 update credential serializer to delete password

### PR DESCRIPTION
**Pull request by cameronperera**
✅ Closes: ICU-5731

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-5731)

## Description

update credential serializer to delete password attribute if empty
string or null allowing the user to update just the username if needed.

### CODE REVIEW

File: addons/api/tests/unit/serializers/credential-test.js [hunk 1]
```diff
@@ -33,6 +33,34 @@ module('Unit | Serializer | credential', function (hooks) {
     });
   });
 
+  test('it serializes username_password type correctly on create no password change', function (assert) {
```
Comment (DhariniJeeva) [inline addons/api/tests/unit/serializers/credential-test.js:36]: Can we rephrase this to improve readability?

File: addons/api/tests/unit/serializers/credential-test.js [hunk 2]
```diff
@@ -64,6 +92,35 @@ module('Unit | Serializer | credential', function (hooks) {
     });
   });
 
+  test('it serializes ssh_private_key type correctly on create no private key or passphrase change', function (assert) {
```
Comment (DhariniJeeva) [inline addons/api/tests/unit/serializers/credential-test.js:95]: same comment as above

**Review by randallmorey — APPROVED** [review]

**Review by lisbet-alvarez — APPROVED** [review]

File: addons/api/addon/serializers/credential.js [hunk 3]
```diff
@@ -28,13 +28,17 @@ export default class CredentialSerializer extends ApplicationSerializer {
     // Remove non-username_password type attributes
     delete serialized.attributes.private_key;
     delete serialized.attributes.passphrase;
+    // Remove password from the payload if null
+    if (!serialized?.attributes?.password)
```
Comment (ZedLi) [inline addons/api/addon/serializers/credential.js:32]: Is this safe? Can someone be allowed to set an empty string as their password? Can the field both be null or undefined or always just one? Curious if we can make these check be a little more specific.

**Review by ZedLi — APPROVED** [review]

File: addons/api/addon/serializers/credential.js (hunk 3 above)
Comment (randallmorey) [reply addons/api/addon/serializers/credential.js:32]: Users cannot unset password.  Probably could add a comment here to that effect.

File: addons/api/tests/unit/serializers/credential-test.js (hunk 1 above)
Comment (cameronperera) [reply addons/api/tests/unit/serializers/credential-test.js:36]: Yes. I will do that now.

File: addons/api/addon/serializers/credential.js (hunk 3 above)
Comment (cameronperera) [reply addons/api/addon/serializers/credential.js:32]: Correct. I can add a comment to explain that here. If the user wants to remove that credential password, they have to delete the whole thing.

**Review by DhariniJeeva — APPROVED** [review]
